### ISSUE TITLE
🐛 BUG: Astro@0.19.1 can't properly handle top-level fragments in `[...rest].astro` files

### ISSUE THREAD

**Issue by okikio** [snowpackjs/astro#1185]
### What package manager are you using?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

Astro@0.19.1 can't properly handle top-level fragments in `[...rest].astro` files, e.g.

```tsx
---
import BaseLayout from "../layouts/BaseLayout";
---
<>
  <BaseLayout>
    <Markdown content={content} />
  </BaseLayout>
</>
```

I get the following error, 
```bash
[06:13:44] [snowpack] Build Result Error: There was a problem with a file build result.
[executing astro] Error: Parse error @:1:1
    at parse (/workspaces/native/node_modules/.pnpm/es-module-lexer@0.3.26/node_modules/es-module-lexer/dist/lexer.cjs:1:402)
    at scanCodeImportsExports (/workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/rewrite-imports.js:21:29)
    at transformEsmImports (/workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/rewrite-imports.js:37:27)
    at Object.transformFileImports (/workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/rewrite-imports.js:108:16)
    at FileBuilder.resolveImports (/workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/build/file-builder.js:116:48)
    at async Object.loadUrl (/workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/commands/dev.js:641:39)
    at async load (/workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/commands/dev.js:217:28)
    at async /workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/ssr-loader/index.js:60:28
    at async Object.importModule (/workspaces/native/node_modules/.pnpm/snowpack@3.8.6/node_modules/snowpack/lib/cjs/ssr-loader/index.js:165:26)
    at async load (file:///workspaces/native/node_modules/.pnpm/astro@0.19.1_typescript@4.4.1-rc/node_modules/astro/dist/runtime.js:90:17)
    at async Server.<anonymous> (file:///workspaces/native/node_modules/.pnpm/astro@0.19.1_typescript@4.4.1-rc/node_modules/astro/dist/dev.js:21:20) {
  idx: 0
}
```

My astro config file is,
```ts
export default {
    // projectRoot: '.',     // Where to resolve all URLs relative to. Useful if you have a monorepo project.
    // pages: './docs', // Path to Astro components, pages, and data
    // dist: './dist',       // When running `astro build`, path to final static output
    // public: './public',   // A folder of static files Astro will copy to the root. Useful for favicons, images, and other files that don’t need processing.
    buildOptions: {
      site: 'https://okikio.github.io/native',  // Your public domain, e.g.: https://my-site.dev/. Used to generate sitemaps and canonical URLs.
      // sitemap: true,      // Generate sitemap (set to "false" to disable)
    },
    devOptions: {
      // port: 3000,         // The port to run the dev server on.
      tailwindConfig: './tailwind.config.cjs', // Path to tailwind.config.js if used, e.g. './tailwind.config.js'
    },
    renderers: ['@astrojs/renderer-solid'],
};
```

### Steps to Reproduce

1. `pnpm i astro@^0.19.1 @astrojs/renderer-solid@^0.1.0 -D`
2. Create an `astro.config.mjs` with 
```ts
export default {
    // projectRoot: '.',     // Where to resolve all URLs relative to. Useful if you have a monorepo project.
    // pages: './docs', // Path to Astro components, pages, and data
    // dist: './dist',       // When running `astro build`, path to final static output
    // public: './public',   // A folder of static files Astro will copy to the root. Useful for favicons, images, and other files that don’t need processing.
    buildOptions: {
      site: 'https://okikio.github.io/native',  // Your public domain, e.g.: https://my-site.dev/. Used to generate sitemaps and canonical URLs.
      // sitemap: true,      // Generate sitemap (set to "false" to disable)
    },
    devOptions: {
      // port: 3000,         // The port to run the dev server on.
      tailwindConfig: './tailwind.config.cjs', // Path to tailwind.config.js if used, e.g. './tailwind.config.js'
    },
    renderers: ['@astrojs/renderer-solid'],
};
```
3. `mkdir src && cd src && mkdir pages && cd pages`
4. `echo "## about" > about.md`
5. `echo "---" > index.astro`
Add this to the file `index.astro` file
```astro
---
import BaseLayout from "../layouts/BaseLayout";
---
<BaseLayout>
  <Markdown content={content} />
</BaseLayout>
```
6. You will then need to create a `src/layouts/BaseLayouts.astro` file, with any random content.
7. `astro dev`
8. Error! I would just expect `<BaseLayout>...</BaseLayout>` to work, unforunately I can't determine if this is because I am using Astro in `[...rest].astro` files
> _**Note**: you have to open [http://localhost:3000](http://localhost:3000) in the browser for the error to appear._

**Comment by jasikpark** [comment]
can you add a title to the issue?

**Comment by okikio** [comment]
Sorry, for that I keep forgetting

**Comment by okikio** [comment]
I was able to create a minimal reproduction on CodeSandbox, https://codesandbox.io/s/astro-top-level-fragments-e4ehd

**Comment by matthewp** [comment]
Is this specific to rest style routes or does this occur in any page? I would think it's a bug that happens regardless of the page.

**Comment by okikio** [comment]
From what I can tell only rest style routes. It still works on ordinary `.astro` pages

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
